middleware/auth.js

```diff
@@ -0,0 +1,36 @@
+const jwt = require("jsonwebtoken");
+
+const asyncHandler = require("./async");
+const ErrorResponse = require("../utils/errorResponse");
+
+// Protect routes
+exports.requireAuth = asyncHandler(async (req, res, next) => {
+  let token;
+  const { authorization } = req.headers;
+
+  if (authorization && authorization.startsWith("Bearer")) {
+    // Set token from Bearer token in header
+    token = authorization.split(" ")[1];
+    // Set token from cookie
+  }
+  // else if (req.cookies.token) {
+  //   token = req.cookies.token;
```

Comment:
- [ ] Either remove this code or add a "TODO:" to help us understand what we want to do here in the future.

Reply:
Done